addons/mail/static/src/js/views/activity/activity_renderer.js

```diff
@@ -49,15 +55,52 @@ var ActivityRenderer = AbstractRenderer.extend({
         var $rows = _.map(this.state.data.res_ids, this._renderRow.bind(this));
         return $('<tbody>').append($rows);
     },
+    /**
+     * @private
+     * @returns {jQueryElement} a <tfoot> element
+     */
+    _renderFooter: function () {
+        return QWeb.render('mail.ActivityViewFooter', {name: _t('Add ') + this.arch.attrs.string});
+    },
     /**
      * @private
      * @returns {jQueryElement} a jquery element <thead>
      */
     _renderHeader: function () {
-        var $tr = $('<tr>')
-                .append($('<th>')) //empty cell for name
-                .append(_.map(this.state.data.activity_types, this._renderHeaderCell.bind(this)));
-        return $('<thead>').append($tr);
+        var self = this;
+        var activityTypeIDs = _.unique(_.flatten(_.map(self.state.data.grouped_activities, function (item) {
+            return _.map(_.keys(item), Number);
+        })));
+        var $cells = _.map(this.state.data.activity_types, function (type) {
+            var $th = $(self._renderHeaderCell(type));
+            if (_.contains(activityTypeIDs, type[0])) {
+                var counts = {planned: 0, today: 0, overdue: 0};
+                _.each(self.state.data.grouped_activities, function (act) {
+                    if (_.contains(_.keys(act), type[0].toString())) {
+                        counts[act[type[0]].state] += 1;
+                    }
+                });
+
+                var progressBar = new ActivityColumnProgressBar(this,
+                    {columnID: type[0], progressBarStates: {}},
+                    {
+                        count: _.reduce(_.values(counts), function (x, y) { return x + y; }),
+                        progressBarValues: {
+                            field: 'activity_state',
+                            colors: {planned: 'success', today: 'warning', overdue: 'danger'},
+                            counts: counts,
+                        }
+                    });
+                progressBar.appendTo($th);
+                progressBar.on('tweak_activity_records', self, self._onTweakActivityRecords.bind(self));
+                progressBar.$counter.remove();
+            }
+            else {
+                $('<div class="o_progress_bar_disabled"/>').appendTo($th);
+            }
+            return $th;
+        });
+        return $('<thead>').append($('<tr>').append($('<th>')).append($cells));
```

Comment:
You should use Qweb templates  more often, instead of creating nodess with jquery

Reply:
Done